### ISSUE TITLE
Inconsistent checking of RBF rules

### ISSUE THREAD

**Issue by LLFourn** [bitcoindevkit/bdk#256]
[BIP125] rules 3 and 4 are not being checked properly according to my understanding:

https://github.com/bitcoindevkit/bdk/blob/c2b2da7601fb570952cfd858f3821b3a4fd87ec9/src/wallet/mod.rs#L730-L746

It should be checking rules 3 and 4 regardless of the fee policy chosen in txbuilder. 

[BIP125]: https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki#Implementation_Details

**Comment by afilini** [comment]
I think we can kinda ignore rule 3 at the moment, since we don't support merging multiple unconfirmed transactions, so this:

> The replacement transaction pays an absolute fee of at least the sum paid by the original transactions.

For us basically means that we should just pay at least the previous fee.

I agree on rule 4 though, the `FeeAmount` branch should check for `*amount < details.fees + _something_`.

And now that I think about it, it's kind of a chicken-and-egg problem: rule 4 says that we should pay at least the size **of the new** transaction, but at this point we don't know it yet. So even the part above where `required_feerate` is computed is wrong.. Not sure what's the best way to handle this, I've gotta think about it a bit more..

**Comment by evanlinjin** [comment]
Coin selection algorithms should take `min_absolute_fee` and `target_feerate` into consideration on the get-go.`bdk_core` will solve this issue, WIP here: https://github.com/LLFourn/bdk_core_staging/pull/21

**Comment by nondiremanuel** [comment]
We discussed it in the call and decided to close this issue. Please comment if you think it should be re-opened

**Comment by notmandatory** [comment]
Per discussion in dev call today this doesn't seem to be fixed yet.

**Comment by LLFourn** [comment]
I volunteer to try and fix RBF API with `bdk_coin_select`. Don't have an ETA yet.

**Comment by notmandatory** [comment]
Thanks @LLFourn , if you can fix with integration of `bdk_coin_select` that'd be great.

**Comment by notmandatory** [comment]
@LLFourn Since we're not going to have have a `bdk_coin_select` crate for 1.0 can I defer this issue to the 2.0 milestone?

**Comment by notmandatory** [comment]
Yes I'd like to make the extracted `bdk_coin_select` one of the priorities for 2.0 so if it's better to fix this together with that I'll update the milestone.

**Comment by LLFourn** [comment]
Sorry ACK I haven't had time to do this work so yes you can push it to 2.0.0. Thanks.

**Comment by ValuedMammal** [comment]
We can wait until we have the `Psbt` in hand, get the fee from the psbt and compute an estimated feerate based on the weight of the unsigned tx and the combined satisfaction weights, then proceed to check rules 3 and 4 by comparing the fee/feerate we calculate with the fee of the previous transaction assuming we're in bumping-fee mode.
